etl/src/main/java/org/pih/hivmigration/etl/sql/HivExamOisMigrator.groovy

```diff
@@ -99,15 +102,29 @@ class HivExamOisMigrator extends ObsMigrator {
               source_encounter_id,
               source_patient_id,
               oi,              
-              comments 
+              comments,
+              form_name,
+              form_version
             )
-            values(?,?,?,?) 
+            values(?,?,?,?,?,?) 
             ''', '''
             select 
                 x.ENCOUNTER_ID as source_encounter_id, 
                 e.patient_id as source_patient_id, 
                 lower(trim(x.oi)) as oi,                  
-                x.COMMENTS
+                x.COMMENTS,
+                case
+                    when (x.encounter_id in (select k.encounter_id from HIV_INTAKE_FORMS k where k.encounter_id=x.encounter_id)) then 'hiv_intake'                     
+                    when (x.encounter_id in (select f.encounter_id from HIV_FOLLOWUP_FORMS f where f.encounter_id=x.encounter_id)) then 'hiv_followup'                     
+                    else null 
+                end as form_name,
+                case
+                    when (x.encounter_id in (select k.encounter_id from HIV_INTAKE_FORMS k where k.encounter_id=x.encounter_id)) then 
+                    ( select form_version from HIV_INTAKE_FORMS where encounter_id=x.encounter_id ) 
+                    when (x.encounter_id in (select f.encounter_id from HIV_FOLLOWUP_FORMS f where f.encounter_id=x.encounter_id)) then 
+                    ( select form_version from HIV_FOLLOWUP_FORMS where encounter_id=x.encounter_id ) 
+                    else null 
+                end as form_version
             from HIV_EXAM_OIS x, HIV_ENCOUNTERS e, hiv_demographics_real d  
             where x.ENCOUNTER_ID = e.ENCOUNTER_ID and e.patient_id = d.patient_id;
```

Comment:
Any reason you didn't just `JOIN` on the `_FORMS` tables? Would simplify the code and might be faster.

Reply:
Yeah, I think a join would work here?

Reply:
Thanks guys! I have updated the query to use joins. Thanks!